src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.netfx.cs

```diff
@@ -790,7 +810,15 @@ private Task SNIWritePacket(SNIPacket packet, out uint sniError, bool canAccumul
 
 #pragma warning restore 420
 
-        internal bool IsValidPacket(PacketHandle packetPointer) => packetPointer != default;
+        internal bool IsValidPacket(PacketHandle packetPointer)
+        {
+            Debug.Assert(packetPointer.Type == PacketHandle.NativePointerType || packetPointer.Type == PacketHandle.NativePacketType, "unexpected packet type when requiring NativePointer");
+            return (
```

Comment:
nit: technically the parenthesis at the beginning here are redundant.
if it were me, I'd consider rewriting this like:
```
return (xyz && abc)
    || (xyz && abc);
```

Reply:
Agreed. I'll make this change in #3355, when the method is ready to merge.

Reply:
I'll try to hold you to it :)